docs: tidy up API headings

petebacondarwin opened the pull request:
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->
```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] angular.io application / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Issue Number: N/A


## What is the new behavior?


## Does this PR introduce a breaking change?
```
[ ] Yes
[ ] No
```

<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

petebacondarwin committed:
build(docs-infra): add @nocollapse tag-def to prevent warning

See https://github.com/angular/angular/blob/master/packages/compiler-cli/src/transformers/nocollapse_hack.ts

petebacondarwin committed:
docs(common): remove legacy `@whatItDoes` tag

petebacondarwin committed:
build(docs-infra): remove legacy jsdoc tag processing

petebacondarwin committed:
docs(common): move `KeyValuePipe` example to `@usageNotes`

petebacondarwin committed:
docs(upgrade): move examples etc into `@usageNotes`

petebacondarwin committed:
docs(router): move examples to `@usageNotes`

petebacondarwin committed:
docs(platform-browser): move examples to `@usageNotes`

petebacondarwin committed:
docs(http): move examples to `@usageNotes`

petebacondarwin committed:
docs(forms): move extended text to @usageNotes

Headings are not allowed in the basic description block.

petebacondarwin committed:
docs(common): move `KeyValuePipe` example to `@usageNotes`

petebacondarwin committed:
docs(core): move headings to `@usageNotes`

mary-poppins commented:
You can preview 14cd104 at https://pr26039-14cd104.ngbuilds.io/.

IgorMinar reviewed (approved):
thanks! can you also make sure that new edits that break these rules also break the build? thanks

IgorMinar commented:
also there are compilation and lint errors that need to be resolved

petebacondarwin committed:
fixup! docs(forms): move extended text to @usageNotes

mary-poppins commented:
You can preview 1136447 at https://pr26039-1136447.ngbuilds.io/.

petebacondarwin committed:
docs(core): remove usage notes from ReflexiveInjector.parent property

Properties are not allowed usage notes, and in this case the example
is so simple it didn't warrant moving it to the overall class documentation.

petebacondarwin committed:
build(docs-infra): allow usage notes on decorator option properties

mary-poppins commented:
You can preview e8834b7 at https://pr26039-e8834b7.ngbuilds.io/.

petebacondarwin committed:
build(docs-infra): fail doc-gen if a content rule fails

mary-poppins commented:
You can preview 775e788 at https://pr26039-775e788.ngbuilds.io/.
